features/keychain/module/keychain.js

```diff
@@ -172,16 +159,78 @@ export class Keychain {
       publicKey = await sodium.privToPub(privateKey)
     }
 
+    return publicKey
+  }
+
+  async exportKey({ seedId, keyId, exportPrivate }) {
+    assert(typeof seedId === 'string', 'seedId must be a string')
+
+    if (exportPrivate) {
+      this.#assertPrivateKeysUnlocked([seedId])
+    }
+
+    const hdkey = this.#getPrivateHDKey({
+      seedId,
+      keyId: new KeyIdentifier(keyId),
+      getPrivateHDKeySymbol: this.#getPrivateHDKeySymbol,
+    })
+    const publicKey = await this.#getPublicKeyFromHDKey({ hdkey, keyId })
+
     const { xpriv, xpub } = hdkey.toJSON()
     return {
       xpub,
       xpriv: exportPrivate ? xpriv : null,
       publicKey,
-      privateKey: exportPrivate ? privateKey : null,
+      privateKey: exportPrivate ? hdkey.privateKey : null,
+    }
+  }
+
+  async getPublicKey({ seedId, keyId }) {
```

Comment:
do we have external consumers that need this or should we make it private for now?

Reply:
I thought it would be nice to have this function so that we don't need to use `exportKey` in places where we know we will never need the private keys or xpubs. Making it private will serve no purpose. I can remove it if it seems overkill.

Reply:
the current one is easy to use though:
```js
const { publicKey } = await keychain.exportKey({ seedId, keyId })
// now we're adding this, which seems super similar
const publicKey = await keychain.getPublicKey({ seedId, keyId })
```

Reply:
The above is exactly the same. Only `getPublicKey` emphasizes we area only interested in the public key. And `exportKey` does cause xpubs to be transported over IPC, even if they are not used afterwards. Also, it is just too easy to add `exportPrivate: true` to `exportKey`. I just think it could improve security and readability.

Reply:
sounds good, let's keep it